src/edu/ucsb/nceas/metacat/dataone/D1AuthHelper.java

```diff
@@ -726,54 +715,72 @@ public boolean isLocalCNAdmin(Session session) throws ServiceFailure {
+        } catch (NotAuthorized na) {
+            return allowed;
         }
-        
-        logMetacat.debug("In isLocalNodeAdmin(), MN authorization for the user " +
-             session.getSubject().getValue());
-        
+        logMetacat.debug("D1AuthHelper.isLocalNodeAdmin(), MN authorization for the user: "
+            + sessionSubjectValue);
+
         Node node = MNodeService.getInstance(request).getCapabilities();
         NodeReference nodeReference = node.getIdentifier();
-        logMetacat.debug("In isLocalNodeAdmin(), Node reference is: " + nodeReference.getValue());
-        
+        logMetacat.debug(
+            "D1AuthHelper.isLocalNodeAdmin(), Node reference is: " + nodeReference.getValue());
+
         Set<Subject> sessionSubjects = AuthUtils.authorizedClientSubjects(session);
-        
+
+        // First, check for a local node subject
         if (nodeType == null || node.getType() == nodeType) {
             List<Subject> nodeSubjects = node.getSubjectList();
             if (sessionSubjects != null) {
-              outer:
+                outer:
                 for (Subject subject : sessionSubjects) {
-                    // check if the session subject is in the node subject list
+                    // Check if the session subject is in the node subject list
                     for (Subject nodeSubject : nodeSubjects) {
-                        logMetacat.debug("In isLocalNodeAdmin(), comparing subjects: " +
-                            nodeSubject.getValue() + " and the user" + subject.getValue());
-                        if ( nodeSubject.equals(subject) ) {
-                            allowed = true; // subject of session == this node's subect
+                        logMetacat.debug("D1AuthHelper.isLocalNodeAdmin(), comparing node subject: "
+                            + nodeSubject.getValue() + " and the session user: "
+                            + subject.getValue());
+                        if (nodeSubject.equals(subject)) {
+                            allowed = true; // subject of session == this node's subject
                             break outer;
                         }
-                    }            
+                    }
+                    // If not, check session subject for a metacat admin
+                    String subjectValue = subject.getValue();
+                    if (subjectValue != null && !subjectValue.isBlank()) {
+                        logMetacat.debug("D1AuthHelper.isLocalNodeAdmin(), checking " + subjectValue
+                            + " for Metacat admin privileges.");
+                        try {
+                            if (AuthUtil.isAdministrator(subjectValue, null)) {
+                                allowed = true;
+                                break outer;
+                            }
+                        } catch (MetacatUtilException mue) {
+                            throw new ServiceFailure("0000", mue.getMessage());
```

Comment:
might be better not to throw an exception here, and instead allow the outer loop to complete, since it may find matches in subsequent loops?

Reply:
This makes sense, nice catch!